@here/harp-mapview/lib/poi/PoiManager.ts

```diff
@@ -458,6 +411,85 @@ export class PoiManager {
         }
     }
 
+    /**
+     * Create and add POI [[TextElement]]s to tile with a series of positions.
+     */
+    private addPoi(
+        tile: Tile,
+        poiGeometry: PoiGeometry,
+        technique: PoiTechnique,
+        techniqueIdx: number,
+        positions: THREE.BufferAttribute
+    ) {
+        const poiTechnique = technique as PoiTechnique;
+        let imageTextureName = poiTechnique.imageTexture;
+        if (typeof poiTechnique.imageTexturePrefix === "string") {
+            imageTextureName = poiTechnique.imageTexturePrefix + imageTextureName;
+        }
+        if (typeof poiTechnique.imageTexturePostfix === "string") {
+            imageTextureName = imageTextureName + poiTechnique.imageTexturePostfix;
+        }
+
+        if (poiGeometry.stringCatalog === undefined) {
+            return;
+        }
+        const poiTableName = poiTechnique.poiTable;
+        let poiName = poiTechnique.poiName;
+
+        for (let i = 0; i < positions.count; ++i) {
+            const x = positions.getX(i);
+            const y = positions.getY(i);
+            const z = positions.getZ(i);
+
+            assert(poiGeometry.texts.length > i);
+            let text = poiGeometry.stringCatalog[poiGeometry.texts[i]];
+            if (text === undefined) {
+                text = "";
+            }
+
+            if (poiGeometry.stringCatalog !== undefined) {
```

Comment:
Could `poiGeometry.stringCatalog` be `undefined` here if on line 434 we exit the function if it is?

Reply:
Again I just moved some stuff to separate function (refactor). Will check it.